totoify_grafeas/totoifylib.py

```diff
@@ -0,0 +1,269 @@
+    if command is None:
+      self.command = ""
+    else:
+      self.command = command
+    # In Grafeas, fields in byproducts and environment of an in-toto link
+    # go within a subfield, "custom_values".
+    if byproducts is None:
+      self.byproducts = {"custom_values": {}}
+    else:
+      self.byproducts = byproducts
+    if environment is None:
+      self.environment = {"custom_values": {}}
+    else:
+      self.environment = environment
+
+  @classmethod
+  def from_dict(cls, dictionary):
+    """Returns a GrafeasLink instance from a dictionary that corresponds to the
+    fields of GrafeasLink. Useful for loading from JSON."""
+    materials = dictionary['materials']
+    products = dictionary['products']
+    command = dictionary['command']
+    byproducts = dictionary['byproducts']
+    environment = dictionary['environment']
+
+    return cls(materials=materials, products=products, command=command,
+               byproducts=byproducts, environment=environment)
+
+  @classmethod
+  def from_link(cls, link):
+    """Performs the conversion from in_toto.models.link.Link to GrafeasLink"""
+    # Convert artifact maps in in-toto links to list of artifact objects as
+    # defined in Grafeas: https://github.com/grafeas/grafeas/blob/master/proto/v1beta1/intoto.proto#L118-L139
+    materials = []
+    for item in link.materials:
+      materials.append({"resource_uri": item,
+                        "hashes": link.materials[item]})
+    products = []
+    for item in link.products:
+      products.append({"resource_uri": item,
+                       "hashes": link.products[item]})
+
+    byproducts = {"custom_values": {}}
+    for key, value in link.byproducts.items():
+      if key == "return-value":
+        # This highlights a special case - in-toto's reference implementations
+        # store return value as an integer while Grafeas allows only strings.
+        # As noted above, Grafeas stores in-toto's byproducts and environment
+        # in a "custom_values" subfield.
+        byproducts["custom_values"][key] = str(value)
+      else:
+        byproducts["custom_values"][key] = value
+
+    environment = {"custom_values": {}}
+    for key, value in link.environment.items():
+      environment["custom_values"][key] = value
+
+    return cls(materials=materials, products=products, command=link.command,
+               byproducts=byproducts, environment=environment)
+
+  def to_link(self, step_name):
+    """Performs conversion from GrafeasLink to in_toto.models.link.Link"""
+    materials = {}
+    products = {}
+    command = []
+    byproducts = {}
+    environment = {}
+
+    for item in self.materials:
+      materials[item["resource_uri"]] = item["hashes"]
+
+    for item in self.products:
+      products[item["resource_uri"]] = item["hashes"]
+
+    command = self.command
+
+    for key, value in self.byproducts.items():
+      if key == "custom_values":
+        continue
+      byproducts[key] = value
```

Comment:
Probably a matter of taste, but to me this seems easier to read:
```suggestion
      if key != "custom_values":
        byproducts[key] = value
```
Same comment applies below in `for key, value in self.environment.items()`-loop.